src/utils/logic.ts

```diff
@@ -46,7 +47,10 @@ export function setActionBooleanProperty(context: LogicContext, action: LogicAct
     const currentValue = get(component, property, false).toString();
     const newValue = action.state.toString();
     if (currentValue !== newValue) {
-        set(component, property, newValue === 'true');
+        const newComponent = fastCloneDeep(component);
```

Comment:
If the logic is triggered, then the properties of the component change.  If do not use cloning, then the original component of the form is changed.
For example, there is a DataGrid component in the form. Inside the DataGrid component, there is a Text Field component with logic, when triggered, the required property for the textField component is added. For the DataGrid Component, the evaluate Process is executed row by row. Therefore, if validate.required = true is set in any row for TextField after triggering the logic, then this property changes the original component and is applied to all following TextField rows.  
Therefore, in this case, cloning of the component should be used so that the original component remains unchanged.